datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1005,6 +1005,7 @@ macro_rules! binary_array_op_dyn_scalar {
             ScalarValue::LargeUtf8(v) => compute_utf8_op_dyn_scalar!($LEFT, v, $OP, $OP_TYPE),
             ScalarValue::Binary(v) => compute_binary_op_dyn_scalar!($LEFT, v, $OP, $OP_TYPE),
             ScalarValue::LargeBinary(v) => compute_binary_op_dyn_scalar!($LEFT, v, $OP, $OP_TYPE),
+            ScalarValue::FixedSizeBinary(_, v) => compute_binary_op_dyn_scalar!($LEFT, v, $OP, $OP_TYPE),
```

Comment:
I think this is the only actual code change

I wrote some tests for it here: https://github.com/apache/arrow-datafusion/pull/6895